test/integration_test/basic_test.go

```diff
@@ -308,31 +321,128 @@ func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
 		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		require.NoError(t, err)
 		cluster.Spec.FeatureGates = map[string]string{"CSI": "false"}
-		cluster, err = ci_utils.UpdateStorageCluster(cluster)
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		logrus.Info("Enable CSI and validate StorageCluster")
+		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		require.NoError(t, err)
+		cluster.Spec.FeatureGates = map[string]string{"CSI": "true"}
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Delete and validate StorageCluster deletion
+		ci_utils.UninstallAndValidateStorageCluster(cluster, t)
+	}
+}
+
+// BasicStorkRegression test includes the following steps:
+// 1. Deploy PX with Stork enabled by default and validate Stork components and images
+// 2. Validate Stork is enabled by default
+// 3. Validate Stork webhook-controller is empty by default
+// 4. Validate Stork hostName is <nil> by default
+// 5. Delete "stork" pods and validate they get re-deployed
+// 6. Delete "stork-scheduler" pods and validate they get re-deployed
+// 7. Enable Stork webhook-controller and validate
+// 8. Disable Stork webhook-controller and validate
+// 9. Remove Stork webhook-controller and validate
+// 10. Enable hotNetwork and validate
+// 11. Disable hostNetwork and validate
+// 12. Remove hostNetwork and valiate
+// 13. Disable Stork and validate Stork components got successfully removed
+// 14. Enabled Stork and validate Stork components and images
+// 15. Delete StorageCluster and validate it got successfully removed
```

Comment:
Let's add a few more cases:
- add a custom stork image - `spec.stork.image`
- remove the custom stork image 
- add a env variable to stork
- change an existing env variable value
- remove an env variable from stork

I am okay if we just create a ticket to track this and add them later.

Reply:
Yes, let add them later, I created a bug for this PTX-4520. I want to cover the P1 validation in the little time we have before we get hit with the releases. We can cover P2 in the later cycle of automation.